src/lucky_cli/spinner.cr

```diff
@@ -0,0 +1,134 @@
+require "colorize"
+
+class LuckyCli::Spinner
+  DELETE_LINE = "\33[2K\r"
+
+  private property frame_index : Int32 = 0
+  private property? started : Bool = false
+  private property last_frame_rendered_at : Time? = nil
+
+  @done_text : String | ColoredString | Nil
+  @start_text : String | ColoredString
+  getter colorize_frame : Proc(String, ColoredString)
+  getter io : IO
+
+  alias ColoredString = Colorize::Object(String)
+
+  def initialize(
+    @start_text,
+    @io = STDERR,
+    @done_text : String | ColoredString | Nil = nil,
+    @colorize_frame = ->(frame : String) { frame.colorize.bold.green }
+  )
+  end
+
+  def self.start(*args, **named_args)
+    spinner = new(*args, **named_args).start
+    yield.tap do |last_value_from_block|
+      spinner.stop
+    end
+  end
+
+  def start : self
+    self.started = true
+    print_spinner
+
+    spawn do
+      while started?
+        if ready_for_next_frame?
+          increment_frame
+          print_spinner
+        end
+        Fiber.yield
```

Comment:
Need this so the current fiber has a chance to run its code. Otherwise this while just blocks the scheduler forever. This took me awhile to figure out :P